File: 0338-counting-bits/0338-counting-bits.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> ans(n+1);
        for(int i = 0; i<=n; i++){
            int temp = i,cnt = 0;
            while(temp!=0){
                if(temp&1) cnt++;
                temp/=2;
            }
            ans[i] = cnt;
        }
        return ans;
    }
};

File: 0338-counting-bits/0338-counting-bits.js
/**
 * @param {number} n
 * @return {number[]}
 */
var countBits = function(n) {
    const ans = new Array(n+1).fill(0);
    for(let i = 1; i<=n; i++){
        ans[Math.floor(i)] = ans[Math.floor(i/2)] + i%2;
    }
    return ans;
};